covenant-signer/config/metrics.go

```diff
@@ -0,0 +1,46 @@
+package config
+
+import (
+	"fmt"
+	"net"
+)
+
+// MetricsConfig defines the server's metric configuration
+type MetricsConfig struct {
+	// IP of the prometheus server
+	Host string `mapstructure:"host"`
+	// Port of the prometheus server
+	Port int `mapstructure:"port"`
+}
+
+type ParsedMetricsConfig struct {
+	Host string
+	Port int
+}
+
+func (cfg *MetricsConfig) Parse() (*ParsedMetricsConfig, error) {
```

Comment:
why the need for both MetricsConfig and ParsedMetricsConfig?

Reply:
We can just validate MatricsConfig

Reply:
Ah I am fan of using as many types as possible i.e when you do `MetricsConfig.Validate()` the information about the validation is lost to all of the consumers of `MetricsConfig ` i.e whatever code receives `MetricsConfig` it does not know whether config was already validated or not. This information lives solely in the head of the developer.

With parsing, any code that receives `ParsedMetricsConfig` knows that config was already validated i.e information about validation is encoded into the type system.